log/handler_glog.go

```diff
@@ -184,20 +186,21 @@ func (h *GlogHandler) Handle(_ context.Context, r slog.Record) error {
 	// If we didn't cache the callsite yet, calculate it
 	if !ok {
 		h.lock.Lock()
+		defer h.lock.Unlock()
```

Comment:
Afaict this will lock the handler even for the h.origin.Handle call in line 206 (203 in original) which means we hold the lock longer than we need to